py/torch_tensorrt/_compile.py

```diff
@@ -332,3 +330,98 @@ def convert_method_to_trt_engine(
         )
     else:
         raise RuntimeError("Module is an unknown format or the ir requested is unknown")
+
+
+def load(file_path: str = "") -> Any:
+    """
+    Load either a Torchscript model or ExportedProgram. Autodetect the type using
+    try, except
+    """
+    try:
+        ts_module = torch.jit.load(file_path)
+        return ts_module
+    except Exception:
+        pass
+
+    try:
+        exp_program = torch.export.load(file_path)
+        return exp_program
+    except Exception:
+        raise ValueError(
+            "The file doesn't correspond to a Torchscript module or ExportedProgram. Please verify the file path."
+        )
```

Comment:
In both `try/except` clauses, can add `logger.debug("...", exc_info=True)`, to show the exception to the user